core/src/test/java/org/apache/calcite/test/RelMetadataTest.java

```diff
@@ -614,6 +614,13 @@ private void checkExchangeRowCount(RelNode rel, double expected, double expected
     checkRowCount(sql, EMP_SIZE + DEPT_SIZE, 0D, 140D);
   }
 
+  @Test void testRowCountUnionDistinct() {
+    final String sql = "select x from (values 'a', 'b') as t(x)\n"
```

Comment:
`select x from values ('a', 'a') UNION DISTINCT select x from values ('a', 'a')` should produce one row right?